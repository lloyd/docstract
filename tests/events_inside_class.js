// docstract supports a somewhat generic notion of "events".  These may
// be associated with a class or the top level file.  An event has a
// name, description, and payload.  That's it.

/** @class EventEmittingClass */

/**
 * @event progress
 * Allows the listener to understand approximately how much of the
 * page has loaded.
 * @payload {number} The percentage (0..100) of page load that is complete
 */
this.eventEmitter._emit("progress", curProgress);

/**
 * @event foo
 * A foo event, duh.
 * @payload {string} A string with lotsa foo inside!
 */
this.eventEmitter._emit("foo", bar);

/** @endclass */
