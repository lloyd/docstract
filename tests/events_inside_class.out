{
  "classes": [
    {
      "events": [
        {
          "desc": "Allows the listener to understand approximately how much of the\npage has loaded.", 
          "name": "progress", 
          "payload": {
            "desc": "The percentage (0..100) of page load that is complete", 
            "type": "number"
          }, 
          "source_lines": [
            7, 
            12
          ]
        }, 
        {
          "desc": "A foo event, duh.", 
          "name": "foo", 
          "payload": {
            "desc": "A string with lotsa foo inside!", 
            "type": "string"
          }, 
          "source_lines": [
            15, 
            19
          ]
        }
      ], 
      "name": "EventEmittingClass", 
      "source_lines": [
        5, 
        6
      ]
    }
  ], 
  "filename": "tests/events_inside_class.js", 
  "module": "events_inside_class"
}
